src/main/java/frc/robot/utils/DeadbandUtils.java

```diff
@@ -0,0 +1,83 @@
+
+package frc.robot.utils;
+
+import edu.wpi.first.wpilibj.Joystick;
+
+public class DeadbandUtils {
+    public static double[] getXYWithDeadband(Joystick joystick, double deadband) {
+        // Negative because joysticks are inverted
+        double xValue = -joystick.getY() * (joystick.getZ() + 1) / 2;
+        double yValue = -joystick.getX() * (joystick.getZ() + 1) / 2;
+
+        // if one of the joystick axes are out of the deadzone, ignore it for both
+        boolean force = Math.abs(xValue) > deadband || Math.abs(yValue) > deadband;
+
+        xValue = applyDeadband(xValue, deadband, 1.0, force);
+        yValue = applyDeadband(yValue, deadband, 1.0, force);
+
+        return new double[] { xValue, yValue };
+    }
+
+    // Copyright (c) FIRST and other WPILib contributors.
+    // Open Source Software; you can modify and/or share it under the terms of
+    // the WPILib BSD license file in the root directory of this project.
+    /**
+     * Returns 0.0 if the given value is within the specified range around zero. The
+     * remaining range
+     * between the deadband and the maximum magnitude is scaled from 0.0 to the
+     * maximum magnitude.
+     *
+     * @param value        Value to clip.
+     * @param deadband     Range around zero.
+     * @param maxMagnitude The maximum magnitude of the input. Can be infinite.
+     * @param force        Skip the deadband check and only scale the value
+     * @return The value after the deadband is applied.
+     */
+    private static double applyDeadband(double value, double deadband, double maxMagnitude, boolean force) {
```

Comment:
This is a modified excerpt of the `MathUtils` class, I added the `force` parameter which disables the deadzone check but keeps the scaling.